src/main/java/uk/gov/hmcts/reform/pbis/EmailCreator.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.hmcts.reform.pbis;
+
+import static uk.gov.hmcts.reform.pbis.model.TemplateFieldNames.FIRST_NAME;
+import static uk.gov.hmcts.reform.pbis.model.TemplateFieldNames.LAST_NAME;
+import static uk.gov.hmcts.reform.pbis.model.TemplateFieldNames.WELCOME_LINK;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.pbis.config.ApplicationConfig;
+import uk.gov.hmcts.reform.pbis.model.EmailTemplateMapping;
+import uk.gov.hmcts.reform.pbis.model.EmailToSend;
+import uk.gov.hmcts.reform.pbis.model.PrivateBetaRegistration;
+
+@Component
+public class EmailCreator {
+
+    private final Map<String, EmailTemplateMapping> emailTemplateMappings;
+
+    @Autowired
+    public EmailCreator(ApplicationConfig applicationConfig) {
+        this.emailTemplateMappings = applicationConfig.getEmailTemplateMappings()
+            .stream()
+            .collect(Collectors.toMap(
+                mapping -> mapping.getService(), Function.identity()
+            ));
+    }
+
+    /**
+     * Converts the given private beta registration into an object representing
+     * email to be sent via notification service.
+     * @param registration Details of private beta registration
+     * @return Complete information about the email to be sent
+     */
+    public EmailToSend createEmailToSend(final PrivateBetaRegistration registration) {
+
+        assertTemplateConfiguredForService(registration.service);
+
+        return new EmailToSend(
+            registration.emailAddress,
+            getTemplateId(registration.service),
+            getFieldsForTemplate(registration),
+            registration.referenceId
+        );
+    }
+
+    private String getTemplateId(final String service) {
+        return emailTemplateMappings.get(service).getTemplateId();
+    }
```

Comment:
looks like this function is only used once

Reply:
also, is this null safe?

Reply:
It's used only once, but the call reads a bit better when it simply says 'give me the template for this service'.

It's safe - the public method starts with a check and if the service is absent, it throws an exception indicating a configuration problem

Reply:
? `emailTemplateMappings` is a map

Reply:
Another reason to inline it then. This private method is null safe. Just happens to never throw NPE because the code using it does the check.

Just noticed that this check (`assertTemplateConfiguredForService`) is also called only once.

Please inline both.

Reply:
There's nothing wrong with a method being called once. It's much more readable when a method's code deals with a certain level of details and you don't have to read through the parts you're not interested in, in order to find out what's happening.

If `createEmailToSend` contained the details of `assertTemplateConfiguredForService` it would just be less readable. Same with `getTemplateId`.

Just like we validate models once and don't check their correctness every time we use them, not all methods have to validate their inputs. Besides, this edge case is covered by unit tests.